enclave/src/openfinex/response_lexer.rs

```diff
@@ -75,6 +77,54 @@ impl ResponseLexer {
     }
 }
 
+/// lex json content as string
+fn get_json<T: Iterator<Item = char>>(iter: &mut Peekable<T>) -> Result<String, String> {
+    let mut json_string = String::new();
+    let mut indentation_level = 0u128;
+
+    // match first curly braces
+    match iter.next() {
+        Some(c) => match c {
+            '{' => {
+                indentation_level = 1;
+                json_string.push(c);
+                Ok(())
+            }
+            _ => Err(format!("Expected '{{' but found {}", c)),
+        },
+        None => Err("Expected '{' but found end of input string".to_string()),
+    }?;
+
+    while let Some(&c) = iter.peek() {
+        if indentation_level < 1 {
+            break;
+        }
+
+        match c {
+            '{' => {
+                indentation_level += 1;
+            }
+            '}' => {
+                if indentation_level > 0 {
+                    indentation_level -= 1;
+                } else {
+                    return Err("Unexpected closing braces '}'".to_string());
+                }
+            }
+            _ => {}
+        }
+
+        json_string.push(c);
+        iter.next();
+    }
+
+    if indentation_level != 0 {
+        return Err("Missing closing braces for JSON string".to_string());
+    }
+
+    Ok(json_string)
+}
```

Comment:
Needed to extend the lexer, since now we have entire JSON objects in the response string